src/frontend/e2e/tests/sample-page.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { expect, Page, test } from "@playwright/test";
+import path from "path";
+import dotenv from "dotenv";
+import fs from "fs";
+import { BasePage } from "../pages/basePage";
+import { getByTestID } from "../utils/selectors";
+
+dotenv.config({ path: path.resolve(`.env.${process.env.NODE_ENV}`) });
+
+const tableHeaders = [
+  "Private ID",
+  "Public ID",
+  "Collection Date",
+  "Lineage",
+  "Upload Date",
+  "Collection Location",
+  "Sequencing Date",
+  "GISAID",
+];
+const api = `${process.env.BASEAPI}/v2/orgs/${process.env.GROUPID}/pathogens/SC2/samples/`;
+const mockData = JSON.parse(
+  fs.readFileSync("e2e/fixtures/sampleList.json") as unknown as string
+);
+let url: string;
+test.describe("Samples page tests", () => {
+  test.beforeAll(async ({}, workerInfo) => {
+    const { baseURL } = workerInfo.config.projects[0].use;
+    url = `${baseURL}/data/samples`;
+  });
+
+  test("Should verify sample list headers", async ({ page }) => {
+    await displaySamplePage(page);
+    const base = new BasePage(page);
+    tableHeaders.forEach(async (header) => {
+      expect((await base.findByText(header)).first()).not.toBeEmpty();
+    });
+  });
+
+  test.only("Should verify sample data", async ({ page, context }) => {
+    await displaySamplePage(page);
+
+    // get the first record so
+    //for validating attributes rendered on UI
+    const sample = mockData.samples[0];
+
+    //create an intercept to stub response with mock data once we get response with status 200
+    await context.route(
```

Comment:
do we need to do this before any test that uses the mock data?

Reply:
It depends. In some cases we can load through beforeAll but tests like filtering will need it per test because the page is being reloaded.
An alternative will be seeding data in backend